include/net/nrf_cloud_defs.h

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF_CLOUD_DEFS_H__
+#define NRF_CLOUD_DEFS_H__
+
+#include <zephyr/toolchain/common.h>
+
+/* Message schemas defined by nRF Cloud:
+ * https://github.com/nRFCloud/application-protocols/tree/v1/schemas
+ */
+
+/* nRF Cloud appID values */
+#define NRF_CLOUD_JSON_APPID_KEY		"appId"
+#define NRF_CLOUD_JSON_APPID_VAL_AGPS		"AGPS"
+#define NRF_CLOUD_JSON_APPID_VAL_PGPS		"PGPS"
+#define NRF_CLOUD_JSON_APPID_VAL_GNSS		"GNSS"
+#define NRF_CLOUD_JSON_APPID_VAL_LOCATION	"GROUND_FIX"
+#define NRF_CLOUD_JSON_APPID_VAL_DEVICE		"DEVICE"
+#define NRF_CLOUD_JSON_APPID_VAL_FLIP		"FLIP"
+#define NRF_CLOUD_JSON_APPID_VAL_BTN		"BUTTON"
+#define NRF_CLOUD_JSON_APPID_VAL_TEMP		"TEMP"
+#define NRF_CLOUD_JSON_APPID_VAL_HUMID		"HUMID"
+#define NRF_CLOUD_JSON_APPID_VAL_AIR_PRESS	"AIR_PRESS"
+#define NRF_CLOUD_JSON_APPID_VAL_AIR_QUAL	"AIR_QUAL"
+#define NRF_CLOUD_JSON_APPID_VAL_RSRP		"RSRP"
+#define NRF_CLOUD_JSON_APPID_VAL_LIGHT		"LIGHT"
+#define NRF_CLOUD_JSON_APPID_VAL_MODEM		"MODEM"
+#define NRF_CLOUD_JSON_APPID_VAL_ALERT		"ALERT"
+
+/* Message type */
+#define NRF_CLOUD_JSON_MSG_TYPE_KEY		"messageType"
+#define NRF_CLOUD_JSON_MSG_TYPE_VAL_CMD		"CMD"
+#define NRF_CLOUD_JSON_MSG_TYPE_VAL_DATA	"DATA"
+#define NRF_CLOUD_JSON_MSG_TYPE_VAL_DISCONNECT	"DISCON"
+
+/* Max length of a NRF_CLOUD_JSON_MSG_TYPE_VAL_DISCONNECT message */
+#define NRF_CLOUD_JSON_MSG_MAX_LEN_DISCONNECT	200
```

Comment:
This looks to be internal to nrf_cloud library. This is maybe a bit more general comment than just this constant but do we want to expose all these define's in a public header?
I'm not sure where to put the limit as sticking strictly to those used outside of this library may not make sense but it looks to me there are some of these implementation specific constants that could be kept private.

Reply:
correct, this value does not need to be in the header.
fixed.

Reply:
Browsing through the list of constants and searching for them from the code base it seems quite many of them are not used outside of nrf_cloud library. Should we make more of them internal to the library possibly in their original place?

Reply:
I agree there could be some that we want to leave into public API even if they are not used in our apps or samples.
However, you've moved `NRF_CLOUD_JSON_MSG_MAX_LEN_DISCONNECT` already. There could be some more that are at the same category. I'll let you judge that.